src/components/ModalBackupWords.js

```diff
@@ -97,19 +100,24 @@ class ModalBackupWords extends React.Component {
    *
    * @param {Object} e Event emitted when button is clicked
    */
-  handlePassword = (e) => {
+  handlePassword = async (e) => {
     e.preventDefault();
     if (this.refs.formPassword.checkValidity() === false) {
       this.setState({ passwordFormValidated: true });
     } else {
       this.setState({ passwordFormValidated: false });
       const password = this.refs.password.value;
-      if (hathorLib.wallet.isPasswordCorrect(password)) {
-        const words = hathorLib.wallet.getWalletWords(password);
+      try {
+        const accessData = await this.props.wallet.storage.getAccessData();
+        const words = hathorLib.cryptoUtils.decryptData(accessData.words, password);
         this.props.updateWords(words);
         this.setState({ passwordSuccess: true, errorMessage: '' });
-      } else {
-        this.setState({ errorMessage: t`Invalid password` });
+      } catch (err) {
+        if (err.errorCode && err.errorCode === hathorLib.ErrorMessages.DECRYPTION_ERROR) {
```

Comment:
`err` may not always exist since we can throw `null` or `undefined`.
The errors that may happen due to the crypto lib should be handled by the wallet-lib so it should come with `errorCode`s but we should be prepared for the exceptional cases, e.g. somehow the object thrown from the crypto lib is undefined or the dispatch from react-redux gives a null error.

I'll change the statement to `err && err.errorCode === ...`.